superset/db_engine_specs/duckdb.py

```diff
@@ -38,9 +45,208 @@
 COLUMN_DOES_NOT_EXIST_REGEX = re.compile("no such column: (?P<column_name>.+)")
 
 
-class DuckDBEngineSpec(BaseEngineSpec):
+class DuckDBURI(
+    util.namedtuple(  # type: ignore
+        "DuckDBURI",
+        [
+            "password",
+            "database",
+            "motherduck_token",
+            "query",
+        ],
+    )
+):
+    drivername = "duckdb"
+
+    def __str__(self) -> str:
+        query = ""
+        if self.motherduck_token:
+            query += f"?motherduck_token={self.motherduck_token}"
+        else:
+            query += "?"
+        query += "&".join([f"{key}={value}" for key, value in self.query.items()])
+        return f"{DuckDBEngineSpec.engine}:///{self.database}{query}"
+
+    @classmethod
+    def from_str(cls, name: str) -> DuckDBURI:
+        pattern = re.compile(
```

Comment:
Not sure if it works for every MotherDuck URL, but you can try using `make_url` from SQLAlchemy here to parse the URL, instead of a regex. Or even `urllib.parse.urlparse`. Regular expressions make me nervous. 😬

Reply:
I'm also not sure if we need this function, let me try to refactor this.

Reply:
I've removed this class altogether and am using `URL` as per your suggestion.